homeassistant/components/abode/sensor.py

```diff
@@ -43,21 +43,15 @@ def __init__(self, data, device, sensor_type):
         self._sensor_type = sensor_type
         self._name = f"{self._device.name} {SENSOR_TYPES[self._sensor_type][0]}"
         self._device_class = SENSOR_TYPES[self._sensor_type][1]
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def device_class(self):
-        """Return the device class."""
-        return self._device_class
-
-    @property
-    def unique_id(self):
-        """Return a unique ID to use for this device."""
-        return f"{self._device.device_uuid}-{self._sensor_type}"
+        self._attr_name = self._name
+        self._attr_device_class = self._device_class
+        self._attr_unique_id = f"{self._device.device_uuid}-{self._sensor_type}"
+        if self._sensor_type == CONST.TEMP_STATUS_KEY:
+            self._attr_unit_of_measurement = self._device.temp_unit
+        if self._sensor_type == CONST.HUMI_STATUS_KEY:
+            self._attr_unit_of_measurement = self._device.humidity_unit
+        if self._sensor_type == CONST.LUX_STATUS_KEY:
+            self._attr_unit_of_measurement = self._device.lux_unit
```

Comment:
This is no longer returning, thus more elif is more efficient now.
```suggestion
        if self._sensor_type == CONST.TEMP_STATUS_KEY:
            self._attr_unit_of_measurement = device.temp_unit
        elif self._sensor_type == CONST.HUMI_STATUS_KEY:
            self._attr_unit_of_measurement = device.humidity_unit
        elif self._sensor_type == CONST.LUX_STATUS_KEY:
            self._attr_unit_of_measurement = device.lux_unit
```